internal/multi_map.go

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	. "github.com/hazelcast/hazelcast-go-client/core"
+	. "github.com/hazelcast/hazelcast-go-client/internal/common"
+	. "github.com/hazelcast/hazelcast-go-client/internal/protocol"
+	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
+	"time"
+)
+
+type MultiMapProxy struct {
+	*proxy
+}
+
+func newMultiMapProxy(client *HazelcastClient, serviceName *string, name *string) *MultiMapProxy {
+	return &MultiMapProxy{&proxy{client, serviceName, name}}
+}
+
+func (mmp *MultiMapProxy) Put(key interface{}, value interface{}) (increased bool, err error) {
+	keyData, valueData, err := mmp.validateAndSerialize2(key, value)
+	if err != nil {
+		return
+	}
+	request := MultiMapPutEncodeRequest(mmp.name, keyData, valueData, ThreadId)
```

Comment:
`ThreadId` should be specific to the routine of execution.

Reply:
Since Golang does not support re-entrant locks anyway, I guess everyone agrees that we could leave it. Ignore my previous comment.